sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_phone_numbers_client.py

```diff
@@ -246,16 +275,151 @@ def get_purchased_phone_number(
+
+        Gets the list of cities or towns with available phone numbers.
+
+        :param country_code: The ISO 3166-2 country/region two letter code, e.g. US. Required.
+        :type country_code: str
+        :param administrative_division: An optional parameter for the name of the state or province
+         in which to search for the area code. e.g. California. Default value is None.
+        :type administrative_division: str
+        :keyword skip: An optional parameter for how many entries to skip, for pagination purposes. The
+         default value is 0. Default value is 0.
+        :paramtype skip: int
+        :return: An iterator like instance of PhoneNumberLocality
+        :rtype:
+         ~azure.core.paging.ItemPaged[~azure.communication.phonenumbers.models.PhoneNumberLocality]
+        :raises ~azure.core.exceptions.HttpResponseError:
+        """
+        return self._phone_number_client.phone_numbers.list_available_localities(
+            country_code,
+            administrative_division=kwargs.pop(
+                "administrative_division", None),
+            accept_language=self._accepted_language,
+            **kwargs
+        )
+
+    @distributed_trace
+    def list_available_offerings(
+        self,
+        country_code,  # type: str
+        **kwargs
+    ):
+        # type: (...) -> ItemPaged[PhoneNumberOffering]
+        """List available offerings of capabilities with rates for the given country/region.
+
+        List available offerings of capabilities with rates for the given country/region.
+
+        :param country_code: The ISO 3166-2 country/region two letter code, e.g. US. Required.
+        :type country_code: str
+        :param phone_number_type: Filter by phoneNumberType, e.g. Geographic, TollFree. Known values
+         are: "geographic" and "tollFree". Default value is None.
+        :type phone_number_type: ~azure.communication.phonenumbers.models.PhoneNumberType
+        :param assignment_type: Filter by assignmentType, e.g. User, Application. Known values are:
+         "person" and "application". Default value is None.
+        :type assignment_type: ~azure.communication.phonenumbers.models.PhoneNumberAssignmentType
+        :keyword skip: An optional parameter for how many entries to skip, for pagination purposes. The
+         default value is 0. Default value is 0.
+        :paramtype skip: int
+        :return: An iterator like instance of PhoneNumberOffering
+        :rtype:
+         ~azure.core.paging.ItemPaged[~azure.communication.phonenumbers.models.PhoneNumberOffering]
+        :raises ~azure.core.exceptions.HttpResponseError:
+        """
+        return self._phone_number_client.phone_numbers.list_offerings(
+            country_code,
+            phone_number_type=kwargs.pop("phone_number_type", None),
+            assignment_type=kwargs.pop("assignment_type", None),
+            **kwargs
+        )
+
+    @distributed_trace
+    def list_available_area_codes(
```

Comment:
Keep in mind that this is the only SDK in which this is not separated into one method for geographic and another one for toll free area codes.